### ISSUE TITLE
Parsing error for functions that receive function types as parameter

### ISSUE THREAD

**Issue by vdrg** [sc-forks/solidity-coverage#198]
Hello, the parser is failing with this:

```
function setStageStartCondition(bytes32 stageId, function() internal view returns(bool) condition) internal {
    // ...
};
```

and this:

```
contract functionThatReceivesAFunction{
    function fn(function() internal view returns(bool));
}
```

The error is the following:
```
SyntaxError: Expected "!=", ")", "0", "<", "<=", "==", ">", ">=", "^", "false", "hex", "mapping", "true", "v", [1-9], comment, end of line, number, string, or whitespace but "f" found. Line: 456, Column: 15
```

**Comment by cgewecke** [comment]
@vdrg Thanks! Will look into this over the weekend unless you are already considering opening a PR on the parser. Is the compilation target [tokenfoundryr/state-matching](https://github.com/tokenfoundry/state-machine)?

**Comment by vdrg** [comment]
Yes, it is for the state machine :smile:  I will try to fix it this weekend, but will have to learn more about how the parser works.
Thanks!

**Comment by cgewecke** [comment]
@vdrg Great!

**Comment by cgewecke** [comment]
@vdrg Thanks for fixing this. Excellent work :) Published with 0.4.11.

**Comment by vdrg** [comment]
Thank you for your work! it is an awesome tool
